kvserver: gate non-voting replica creation behind new cluster version

aayushshah15 opened the pull request:
This PR makes it such that non-voting replicas cannot be created
before 21.1

Release justification: gating new functionality behind a new cluster
version

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61112)
<!-- Reviewable:end -->

aayushshah15 committed:
kvserver: gate non-voting replica creation in the replicateQueue

This commit makes it such that non-voting replicas cannot be created
before 21.1

Release justification: gating new functionality behind a new cluster
version

Release note: None

aayushshah15 committed:
sql,config: gate `num_voters` and `voter_constraints` on cluster version

This commit gates the new zone config attributes introduced in
version.

Release justification: gating new functionality behind 21.1
Release note: None

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61112#-:-MUPNFMg2Z3c5WeBmApv:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

aayushshah15 commented:
TFTR!

bors r+
